pkg/plugin/v3/scaffolds/internal/templates/main.go

```diff
@@ -44,30 +44,27 @@ func (f *Main) SetTemplateDefaults() error {
 		f.Path = filepath.Join(defaultMainPath)
 	}
 
+	flags := flagsFragment
+	options := optionsFragment
+	componentConfig := ""
+	if f.ComponentConfig {
+		flags = ""
+		options = ""
+		componentConfig = ".AndFromOrDie(ctrl.ConfigFile())"
+	}
+
 	f.TemplateBody = fmt.Sprintf(mainTemplate,
 		file.NewMarkerFor(f.Path, importMarker),
 		file.NewMarkerFor(f.Path, addSchemeMarker),
+		flags,
+		options,
+		componentConfig,
```

Comment:
You should not template in `SetTemplateDefaults`. Use `{{ if .ComponentConfig }` conditionals to template in the template body.